### PR TITLE
feat(stepper): L3-7123 Create progress component (Part 1)

### PR DESCRIPTION
**Pull request by smarks8**
**Jira ticket**

[L3-7123](https://phillipsauctions.atlassian.net/browse/L3-7123)

**Screenshots**
image
image
image

**Figma link**

 [Figma Link](https://www.figma.com/design/kSxOhnqIhilZ9hIJd3bPgP/RW-Registration?node-id=2996-50543&m=dev)
 [Figma Link 2](https://www.figma.com/design/kSxOhnqIhilZ9hIJd3bPgP/RW-Registration?node-id=2529-23889&m=dev)

**Summary**

- Add progress component that will be used in the top of the new upcoming progress wizard
- Splitting the PRs up to make more digestible 

**Change List (describe the changes made to the files)**

This pull request introduces a new `ProgressIndicator` component, providing a visual representation of multi-step progress. It includes the component implementation, styling, tests, and Storybook stories for usage examples. The most important changes are outlined below:

### Component Implementation:
* Added the `ProgressIndicator` component in `src/components/ProgressIndicator/ProgressIndicator.tsx`, supporting configurable steps, current step, optional labels, and styling via class names. It uses the `@radix-ui/react-progress` library and includes accessibility attributes for better usability.

### Styling:
* Introduced a new SCSS file, `_progressIndicator.scss`, to define the styles for the `ProgressIndicator` component, including step circles, connectors, and optional labels with responsive design.

### Testing:
* Added comprehensive unit tests in `ProgressIndicator.test.tsx` to validate the component's behavior under various scenarios, such as rendering steps, handling edge cases, and ensuring accessibility attributes are correctly applied.

### Storybook Integration:
* Created Storybook stories in `ProgressIndicator.stories.tsx` to showcase the `ProgressIndicator` component with examples like a playground and labeled steps. Configured controls for interactive testing of props.

### Export and Index Updates:
* Exported the `ProgressIndicator` component from `index.ts` and updated the main `src/index.ts` file to include the new component in the library's exports. [[1]](diffhunk://#diff-014109879d7ae420f25894523e15eb4ab5384f783f246c105add677188398595R1) [[2]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R101)

**Acceptance Test (how to verify the PR)**

- See in storybook that figma matches

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-7123]: https://phillipsauctions.atlassian.net/browse/L3-7123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/ProgressIndicator/ProgressIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      if (value >= steps) {
+        return steps;
+      }
+      return value;
+    };
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        <Progress.Root value={getValue(current)} max={steps} aria-label="Progress">
+          <div className={`${baseClassName}__steps`} aria-label="Progress steps">
```
Comment (davidicus) [inline src/components/ProgressIndicator/ProgressIndicator.tsx:55]: Same comment about aria-label

File: src/components/ProgressIndicator/ProgressIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      if (value >= steps) return steps;
+      return value;
+    };
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        <Progress.Root value={getValue(current)} max={steps} aria-label={ariaLabelProgress}>
+          <div className={`${baseClassName}__steps`} aria-label={ariaLabelSteps}>
+            {Array.from({ length: steps }).map((_, index) => {
+              const isComplete = current > index + 1;
```
Comment (scottdickerson) [inline src/components/ProgressIndicator/ProgressIndicator.tsx:84]: could set a `const stepNumber = index + 1` to prevent the repetition below

File: src/components/ProgressIndicator/ProgressIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                  ? ariaLabelStepCurrent(index + 1)
+                  : ariaLabelStepNotStarted(index + 1);
+
+              return (
+                <Fragment key={props.labels ? props.labels[index] : index}>
+                  <div
+                    className={`${baseClassName}__item`}
+                    aria-current={isCurrent ? 'step' : undefined}
+                    aria-label={stepAriaLabel}
+                    data-testid="progress-step"
```
Comment (scottdickerson) [inline src/components/ProgressIndicator/ProgressIndicator.tsx:98]: if they were trying to use testid wouldn't we need this to be unique for each step?

File: src/components/ProgressIndicator/ProgressIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    const getValue = (value: number) => {
+      if (value <= 0) return 0;
+      if (value >= steps) return steps;
+      return value;
+    };
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        <Progress.Root value={getValue(current)} max={steps} aria-label={ariaLabelProgress}>
+          <div className={`${baseClassName}__steps`} aria-label={ariaLabelSteps}>
```
Comment (scottdickerson) [inline src/components/ProgressIndicator/ProgressIndicator.tsx:82]: should this be an `<ol` and `<li`?

**Review by DaveMcNamara — APPROVED** [review]
